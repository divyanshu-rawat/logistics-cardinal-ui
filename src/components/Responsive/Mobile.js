import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from 'styled-components';
import Responsive from 'react-responsive';

export const Mobile = ({ theme, ...props }) => (
  <Responsive
    minWidth={theme.breakpoints.mobile.minWidth}
    maxWidth={theme.breakpoints.mobile.maxWidth}
    {...props}
  />
);

Mobile.propTypes = {
  theme: PropTypes.object.isRequired,
};

export default withTheme(Mobile);
