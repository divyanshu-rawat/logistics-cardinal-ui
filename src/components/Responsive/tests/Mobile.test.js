import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Mobile } from '../Mobile';

import Themes from '../../../themes';

const props = {
  theme: Themes.rooster,
  someOtherProp: true,
};

describe('<Mobile />', () => {
  it('should render the component', () => {
    const tree = toJson(shallow(<Mobile {...props}>Test...</Mobile>));
    expect(tree).toMatchSnapshot();
  });
});
