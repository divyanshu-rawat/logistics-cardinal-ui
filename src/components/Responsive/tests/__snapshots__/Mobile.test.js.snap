// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mobile /> should render the component 1`] = `
<MediaQuery
  maxWidth={768}
  minWidth={0}
  someOtherProp={true}
  values={Object {}}
>
  Test...
</MediaQuery>
`;
